### PR TITLE
Adding connection-info kcat support in aiven-client

### PR DESCRIPTION
**Pull request by ftisiot**
# About this change: What it does, why it matters

Kafkacat is the old name of the tool now named [kcat](https://github.com/edenhill/kcat), the PR adds the functions to retrieve the kcat call, we'll be then able to deprecate the `kafkacat` one and remove it safely some time in the future

### CODE REVIEW

**Comment by rominf** [comment]
Converted to draft since tests do not pass to prevent accidental merge.

**Comment by ftisiot** [comment]
Hi! In order to test this please find a Aiven for Kafka service and execute

```
avn service connection-info kcat  [SERVICE_NAME]
```

The output should be similar to 

```
kcat -b HOST:PORT -X security.protocol=SSL -X ssl.ca.location=ca.pem -X ssl.key.location=service.key -X ssl.certificate.location=service.crt
```

If the Aiven for Apache Kafka uses SASL authentication you can use the `-a sasl` flag and you should get something like

```
kcat -b HOST:PORT -X security.protocol=SASL_SSL -X ssl.ca.location=ca.pem -X sasl.mechanisms=SCRAM-SHA-256 -X sasl.username=avnadmin -X sasl.password=PASSWORD
```

**Review by rominf — CHANGES_REQUESTED** [review]
Please add user-friendly message in case Kafka has SASL disabled.

Now it's not user-friendly:
```
avn service connection-info kcat -a sasl [service-without-sasl]
ERROR   command failed: ConnectionInfoError: Could not find connection information with filters route=dynamic, privatelink_connection_id=<object object at 0x7f2a939b4c20>, kafka_authentication_method=sasl
```

File: aiven/client/cli.py [hunk 1]
```diff
@@ -1343,6 +1343,7 @@ def _get_store_from_args(self) -> Store:
             store = Store.skip
         return store
 
+    # This method should be deprecated and removed over time. Check service__connection_info__kcat
```
Comment (HelenMel) [inline aiven/client/cli.py:1346]: I would rather use `warnings.warn("Use the kcat method instead", DeprecationWarning)`

**Comment by HelenMel** [comment]
Good PR. I have one minor comment

File: aiven/client/cli.py (hunk 1 above)
Comment (ftisiot) [reply aiven/client/cli.py:1346]: thanks!

File: aiven/client/connection_info/kafka.py [hunk 2]
```diff
@@ -88,19 +104,25 @@ def from_service(
...
+                route=route,
+                privatelink_connection_id=privatelink_connection_id,
+                kafka_authentication_method="sasl",
+            )
+            
+            user = find_user(service, username)
+            if "password" not in user:
+                raise ConnectionInfoError(f"Could not find password for username {username}")
+            return cls(host=info["host"], port=info["port"], username=username, password=user["password"])
+        except ConnectionInfoError:
```
Comment (rominf) [inline aiven/client/connection_info/kafka.py:118]: If the password is missing, then you'll get `SASL auth...` error, which is wrong. I think the better way is to check user config via separate call to understand if SASL auth is enabled.

**Review by rominf — CHANGES_REQUESTED** [review]

File: aiven/client/connection_info/kafka.py (hunk 2 above)
Comment (ftisiot) [reply aiven/client/connection_info/kafka.py]: gotcha, doing for both SSL and SASL methods

File: aiven/client/connection_info/kafka.py [hunk 3]
```diff
@@ -52,7 +68,15 @@ def from_service(
             raise ConnectionInfoError(
                 "Cannot format kafka connection info for service type {service_type}".format_map(service)
             )
-
+        try:
+            find_component(
+                service["components"],
+                kafka_authentication_method="certificate",
+            )
```
Comment (rominf) [inline aiven/client/connection_info/kafka.py:75]: You have two `find_component` calls with different arguments with the same `kafka_authentication_method="certificate"`. Why not keep just one?

Comment (ftisiot) [reply aiven/client/connection_info/kafka.py:75]: I believe a standalone `find_component` checking only the `kafka_authentication_method` checks if the authentication method is enabled, while the `find_component` checking both `kafka_authentication_method` and `route` checks that the authentication method is enabled in the route specified?

Comment (rominf) [reply aiven/client/connection_info/kafka.py:75]: Ah, yeah, you are right. Thanks.

**Review by rominf — APPROVED** [review]
LTGM. I'll let @HelenMel check this before merging.

**Comment by ftisiot** [comment]
@HelenMel could you please double check that everything is ok from your side?

**Review by HelenMel — APPROVED** [review]

**Pull request merged**